src/Hl7.Fhir.Core/ElementModel/TypedElementExtensions.cs

```diff
@@ -6,14 +6,17 @@
  * available at https://raw.githubusercontent.com/FirelyTeam/firely-net-sdk/master/LICENSE
  */
 
+#nullable enable
+
 using Hl7.Fhir.Introspection;
 using Hl7.Fhir.Model;
 
 namespace Hl7.Fhir.ElementModel
 {
     public static class TypedElementExtensions
     {
-        public static ITypedElement ToTypedElement(this Base @base, string rootName = null) =>
-            new PocoElementNode(ModelInspector.ForAssembly(@base.GetType().Assembly), @base, rootName: rootName);
+        public static ITypedElement ToTypedElement(this Base @base, string? rootName = null)
+            => new PocoElementNode(ModelInspectorExtensions.ForInstance(@base), @base, rootName: rootName);
```

Comment:
Have we carefully tracked where we have been using `ForAssembly()` and it should really be `ForInstance()`?

Reply:
We decided to remove `ForInstance`.